wsmaster/che-core-api-workspace/src/test/java/org/eclipse/che/api/environment/server/compose/EnvironmentContextTest.java

```diff
@@ -0,0 +1,171 @@
+    }
+
+    @DataProvider
+    public Object[][] correctContentTestData() {
+        return new Object[][] {
+            // dictionary type environment
+            {"services: \n"
+             + " dev-machine: \n"
+             + "  image: codenvy/ubuntu_jdk8\n"
+             + "  environment:\n"
+             + "    RACK_ENV: development\n"
+             + "    SHOW: 'true'",
+             ImmutableMap.of("RACK_ENV", "development",
+                             "SHOW", "true")
+            },
+
+            // array type environment
+            {"services: \n"
+             + " dev-machine: \n"
+             + "  image: codenvy/ubuntu_jdk8\n"
+             + "  environment:\n"
+             + "   - MYSQL_ROOT_PASSWORD=root\n"
+             + "   - MYSQL_DATABASE=db",
+             ImmutableMap.of("MYSQL_ROOT_PASSWORD", "root",
+                             "MYSQL_DATABASE", "db")
+            },
+
+            // empty environment
+            {"services: \n"
+             + " dev-machine: \n"
+             + "  image: codenvy/ubuntu_jdk8\n"
+             + "  environment:",
+             ImmutableMap.of()
+            },
+
+            // dictionary format, value of variable is empty
+            {"services: \n"
+             + " dev-machine: \n"
+             + "  image: codenvy/ubuntu_jdk8\n"
+             + "  environment:\n"
+             + "   MYSQL_ROOT_PASSWORD: ",
+             ImmutableMap.of("MYSQL_ROOT_PASSWORD", "")
+            },
+
+            // dictionary format, value of variable contains colon sign
+            {"services: \n"
+             + " dev-machine: \n"
+             + "  image: codenvy/ubuntu_jdk8\n"
+             + "  environment:\n"
+             + "   VAR : val:1",
+             ImmutableMap.of("VAR", "val:1")
+            },
+        };
+    }
+
+    @Test(dataProvider = "incorrectContentTestData")
+    public void shouldThrowError(String content, String errorPattern) throws ServerException {
+        try {
+            parser.parse(content, "application/x-yaml");
+        } catch (IllegalArgumentException e) {
+            assertTrue(e.getMessage().matches(errorPattern),
+                       format("Actual error message \"%s\" doesn't match regex \"%s\" for content \"%s\"",
+                              e.getMessage(),
+                              errorPattern,
+                              content));
+            return;
+        }
+
+        fail(format("Content \"%s\" should throw IllegalArgumentException", content));
+    }
+
+    @DataProvider
+    public Object[][] incorrectContentTestData() {
+        return new Object[][] {
+            // unsupported type of environment
+            {"services: \n"
+             + " dev-machine: \n"
+             + "  image: codenvy/ubuntu_jdk8\n"
+             + "  environment:\n"
+             + "   true",
+             "Parsing of environment configuration failed. Unsupported type 'class java.lang.Boolean'.*(?s).*"
```

Comment:
Why do you need `.*` before and after `(?s)`. after is enough I think. Mb you mean `\\.(?s).*`